### ISSUE TITLE
[ENHANCEMENT] Docs: Llama_Index IngestionPipeline and Transformations support or example

### ISSUE THREAD

**Issue by chrishart0** [Arize-ai/phoenix#2156]
**Is your feature request related to a problem? Please describe.**
Recently LlamaIndex released [IngestionPipelines](https://docs.llamaindex.ai/en/stable/module_guides/loading/ingestion_pipeline/root.html), which are essentially collections of LlamaIndex [transformations](https://docs.llamaindex.ai/en/stable/module_guides/loading/ingestion_pipeline/transformations.html#transformations). These transformation are tasks such as `SummaryExtractor` and `TitleExtractor` which make LLM calls to write summaries of a node or create a title for an entire document. I would like to use pheonix to monitor this process the same way I can making retrievals. These various transformations are making calls to the LLM, those calls should be traceable. I spent some time trying to get this data into pheonix but was not able to figure it out. 

**Describe the solution you'd like**
I would like documentation on how to trace LlamaIndex IngestionPipelines and Transformations. If this is not possible, I would like the functionality implemented. Examples of this process would be highly useful for anyone doing data ingestion at any scale.

**Describe alternatives you've considered**
I have considered that I missed how to do this, if so, please let me know and I am willing to contribute back to the docs. 

**Additional context**
Here is some of my code from my ingestion script:

```
from llama_index.ingestion import IngestionPipeline, IngestionCache
from llama_index.text_splitter import SentenceSplitter
from llama_index.extractors import TitleExtractor
from llama_index.embeddings import HuggingFaceEmbedding
from llama_index.storage.docstore import (
    SimpleDocumentStore
)
from llama_index.extractors import (
    SummaryExtractor,
    QuestionsAnsweredExtractor,
)

callback_handler = OpenInferenceTraceCallbackHandler()
...

llm = Ollama(
    model="mixtral",
    base_url="http://0.0.0.0:11435",
    request_timeout=500,
    context_window=8192
)
service_context = ServiceContext.from_defaults(
    llm=llm, 
    embed_model="local",
    callback_manager=CallbackManager(handlers=[callback_handler]),
)

...

pipeline = IngestionPipeline(
    cache=IngestionCache(

    ),
    transformations=[
        SentenceSplitter(chunk_size=200, chunk_overlap=0),
        SummaryExtractor(summaries=["prev", "self", "next"], llm=llm),
        TitleExtractor(llm=llm),
        QuestionsAnsweredExtractor(llm=llm, questions=3),
        HuggingFaceEmbedding(
            model_name="sentence-transformers/all-MiniLM-L6-v2",
            callback_manager=CallbackManager(handlers=[callback_handler])
        ),
    ],
    vector_store=qdrant_vector_store,
    docstore=SimpleDocumentStore(),
)

pipeline.run(documents=documents, show_progress=True)
```

**Comment by chrishart0** [comment]
Thank you Dosubot, your suggestions didn't fix my issue, but it was a very thorough and well written response. Very cool.

To whatever human reads this, I went through line by line review Dosubot's code, the only thing which changed is I now has `callback_manager = CallbackManager(handlers=[callback_handler])` defined, but I do not use it anywhere. 

I am thinking that for tracing to work each of the transformations, such as `SummaryExtractor`, would need some sort of callback provided that pheonix could drop into. Since those aren't provided, I am going to assume there isn't a way to do this without llama_index code changes, maybe some else knows?

For some Llama_index methods you can pass in a service context which contains an llm and callback_manager, but these transformers just take an LLM.

**Comment by mikeldking** [comment]
@chrishart0 thank you so much for the enhancement request! For some reason this ticket completely slipped our triaging process. Will get back to you on the scoping and priority of this. LlamaIndex has moved to `instrumentors` and we will probably have to scope it as part of that work as the callbacks are being deprecated. We are working closely with that team and should be able to give you and update once we are unblocked.

**Comment by mikeldking** [comment]
Hi @chrishart0 just an update that we released 2.0 of our llama-index instrumentor that might solve some things. https://github.com/Arize-ai/openinference/tree/main/python/instrumentation/openinference-instrumentation-llama-index

We will dive in a bit more and let you know!
